### DISCUSSION TITLE
Polar plots

### DISCUSSION THREAD

**Discussion by georgweiss** [ControlSystemStudio/phoebus#1874] [General]
I have users asking for polar plot support in Phoebus. Between a third party library and extending current plot APIs I would probably lean towards the first. But as this would most probably be a major undertaking either way I'd like to collect opinions, pros and cons.

**Comment by kasemir** [comment]
The first problem might be defining how the plot should work and what the data needs to look like.
Would it be in units of degrees or radians?
A third party library would very likely result in axes that look different from the existing plots, perform drawing on the UI thread and add many more JavaFX scene elements compared to just one 'image' for existing plots. Plus each new widget needs to be maintained #1389.

Since it is quite likely that you need to somehow transform the original data for representation in the plot, you might as well use a script to transform it into plain X/Y waveforms for the existing XY Plot:

[polar.bob.zip](https://github.com/ControlSystemStudio/phoebus/files/6343558/polar.bob.zip)

![Screen Shot 2021-04-20 at 8 37 05 AM](https://user-images.githubusercontent.com/1932421/115397011-9a9b0280-a1b3-11eb-8472-f67ffd6c6d24.png)

This lacks support for a polar-aware cursor, you'll just get X/Y coordinates from the crossbar, but does give a basic representation of the data.

**Reply by georgweiss** [reply]
To clarify, users are asking for this:
![SpecifyLineColorForPolarPlotExample_01](https://user-images.githubusercontent.com/35602960/115399237-46f6dc00-a1e8-11eb-980e-fb52d612eb2a.png)
